include/dlaf/matrix/views.h

```diff
@@ -0,0 +1,136 @@
+  }
+
+  // TODO this is used for determing first, but it's not a solution
+  LocalTileIndex begin() const noexcept {
+    return tile_begin_;
+  }
+
+protected:
+  Distribution dist_;
+  GlobalElementIndex offset_e_;
+  GlobalTileIndex offset_tile_;
+
+  LocalTileIndex tile_begin_, tile_end_;
+};
+
+struct SubMatrixView : public IndexHelper {
+  SubMatrixView(Distribution dist, GlobalElementIndex offset_e) : IndexHelper(dist, offset_e) {
+    if (dist.size().isEmpty())
+      return;
+
+    tile_end_ = indexFromOrigin(dist_.localNrTiles());
+
+    has_subtile_ = dist_.rankGlobalTile<Coord::Row>(offset().row()) == dist_.rankIndex().row() ||
+                   dist_.rankGlobalTile<Coord::Col>(offset().col()) == dist_.rankIndex().col();
+  }
+
+  matrix::SubTileSpec operator()(const LocalTileIndex& index) const {
+    // TODO index must be valid inside the range
+    SizeType i_sub = 0, j_sub = 0;
+
+    if (has_subtile_) {
+      const auto ij = dist_.globalTileIndex(index);
+
+      if (ij.row() == offset().row())
+        i_sub += dist_.tileElementFromGlobalElement<Coord::Row>(offset_e_.row());
+      if (ij.col() == offset().col())
+        j_sub += dist_.tileElementFromGlobalElement<Coord::Col>(offset_e_.col());
+    }
+    const TileElementIndex offset_sub(i_sub, j_sub);
+
+    const TileElementSize size_tile = dist_.tileSize(dist_.globalTileIndex(index));
+    const TileElementSize size_sub(size_tile.rows() - offset_sub.row(),
+                                   size_tile.cols() - offset_sub.col());
+    return {offset_sub, size_sub};
+  }
+
+private:
+  bool has_subtile_;
+};
+
+struct SubPanelView : public IndexHelper {
+  SubPanelView(Distribution dist, GlobalElementIndex offset_e, const SizeType width)
+      : IndexHelper(dist, offset_e), width_(width) {
+    if (dist.size().isEmpty())
+      return;
+
+    i_sub_offset_ = offset_e_.row() % dist_.blockSize().rows();
+    j_sub_offset_ = offset_e_.col() % dist_.blockSize().cols();
+
+    tile_end_ = LocalTileIndex(dist_.localNrTiles().rows(), tile_begin_.col() + 1);
+
+    has_top_tile_ = dist_.rankGlobalTile<Coord::Row>(offset().row()) == dist_.rankIndex().row();
+  }
+
+  // TODO we can add an helper c'tor that given a GlobalTileIndex it creates a panel tile-wide
+
+  matrix::SubTileSpec operator()(const LocalTileIndex& index) const {
+    // TODO index must be valid inside the range
+    DLAF_ASSERT_MODERATE(index.isIn(dist_.localNrTiles()), index, dist_.localNrTiles());
+
+    const TileElementIndex offset_sub((has_top_tile_ && index == tile_begin_) ? i_sub_offset_ : 0,
+                                      j_sub_offset_);
+
+    const TileElementSize size_tile = dist_.tileSize(dist_.globalTileIndex(index));
+    const TileElementSize size_sub(size_tile.rows() - offset_sub.row(),
+                                   std::min(width_, size_tile.cols() - offset_sub.col()));
+    return {offset_sub, size_sub};
+  }
+
+  SizeType width() const noexcept {
```

Comment:
What exactly does this represent? Without understanding this fully I'm wondering why it's not `cols` so: what makes it not `cols`?

Reply:
It is actually the number of columns, so it's a good point to keep it coherent with the rest of the library.

Taking the chance to point put one thing: at the moment `SubPanelView` is not generic for Row/Col, but it is just a column one.